### PR TITLE
[Docs] Add "Using Java Builders" section

### PR DESCRIPTION
**Pull request by tlrx**
The current "Building Queries" and "Building Aggregations" pages are
located under the "Supported Apis" section because they are linked to
the "Search API" page.

It should instead be in a dedicated section: this commit adds a new
"Using Java Builders" section and renames few filenames in favor of
more meaningful names.

### CODE REVIEW

**Comment by cbuescher** [comment]
@tlrx great, let me just build those and see how it looks like

File: docs/java-rest/high-level/java-builders.asciidoc [hunk 1]
```diff
@@ -0,0 +1,26 @@
+[[java-rest-high-java-builders]]
+== Using Java Builders
+
+The Java High Level REST Client depends on the Elasticsearch core project which provides
+two types of Java `Builders` objects:
```
Comment (cbuescher) [inline docs/java-rest/high-level/java-builders.asciidoc]: I don't know if we should say there are only two types of builders, after all search also has Suggestion builder, Sort, Highlightingbuilders etc... We don't need to mention them all here but maybe we can say these two are the most important ones?

File: docs/java-rest/high-level/java-builders.asciidoc [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+Query Builders::
+
+The query builders are used to create `QueryBuilder` objects which can later be set as the
```
Comment (cbuescher) [inline docs/java-rest/high-level/java-builders.asciidoc]: I think this is a bit confusing, at least to me. Query builders "are" the the implementations of the "QueryBuilder" interface, they don't create them but rather later create the Lucene queries on the shards. We don't need to mention that here, maybe better say that the query builders are used to describe the full query and implement the "QueryBuilder" interface, that is used in the SearchSourceBuilder? Or something else along those lines.

File: docs/java-rest/high-level/java-builders.asciidoc [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+query parameter of `SearchSourceBuilder` objects. The <<java-rest-high-document-search-request-building-queries, Search Request>>
+page shows an example of how to build a full search request using `SearchSourceBuilder` and
+`QueryBuilder` objects. The <<java-rest-high-query-builders, Building Search Queries>> page
+gives a list of all available search queries with their corresponding `QueryBuilder` objects and
+`QueryBuilders` helper methods.
+
+Aggregation Builders::
+
+Similarly to query builders, the aggregation builders are used to create `AggregationBuilder` objects
```
Comment (cbuescher) [inline docs/java-rest/high-level/java-builders.asciidoc]: Same here. I like the rest of the summary here though.

**Review by cbuescher — COMMENTED** [review]
@tlrx thanks, the restructuring looks good to me, I left a few comment on the new "Using Java Builders" section

**Comment by tlrx** [comment]
@cbuescher Thanks for your feedback. I agree with all your points so I reworded things a bit. Please let me know what you think, and feel free to make more comments if needed :)

**Review by cbuescher — APPROVED** [review]
@tlrx thanks a lot, great change. It reads really good now.

**Comment by tlrx** [comment]
Thanks @cbuescher

**Pull request merged**